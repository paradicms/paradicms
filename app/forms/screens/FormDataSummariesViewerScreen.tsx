import * as React from "react";
import {useEffect, useState} from "react";
import {FormStackScreenProps} from "../navigation/FormStackScreenProps";
import {useFormDataApi} from "../hooks/useFormDataApi";
import {FormDataSummariesViewer} from "../components/FormDataSummariesViewer";
import {FormDataSummary} from "@paradicms/shacl-forms";

export const FormDataSummariesViewerScreen: React.FunctionComponent<FormStackScreenProps<
  "FormDataSummariesViewerScreen"
>> = ({navigation, route}) => {
  const {formDataApi} = useFormDataApi();
  const [formDataSummaries, setFormDataSummaries] = useState<
    readonly FormDataSummary[] | null
  >(null);

  useEffect(() => {
    if (formDataApi) {
      formDataApi.getFormDataSummaries().then(setFormDataSummaries);
    }
  }, [formDataApi]);

  if (!formDataSummaries) {
    return null;
  }

  return (
    <FormDataSummariesViewer
      formDataSummaries={formDataSummaries}
      onSelectFormDataSummary={formDataSummary =>
        navigation.push("FormDataViewerScreen", {
          formData: formDataSummary,
        })
      }
    />
  );
};
