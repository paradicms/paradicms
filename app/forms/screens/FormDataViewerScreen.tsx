import * as React from "react";
import {FormStackScreenProps} from "../navigation/FormStackScreenProps";
import {FormDataViewer} from "@paradicms/react-native-shacl-forms";
import {useFormData} from "../hooks/useFormData";
import {CenteredView} from "../components/CenteredView";

export const FormDataViewerScreen: React.FunctionComponent<FormStackScreenProps<
  "FormDataViewerScreen"
>> = ({navigation, route}) => {
  const {error, formData} = useFormData(route.params.formData.id);

  if (error) {
    navigation.navigate("FormErrorViewerScreen", {
      ...route.params,
      formNodeData: null,
      formNodeTypeData: null,
      error,
    });
    return null;
  }

  if (!formData) {
    return null;
  }

  return (
    <CenteredView>
      <FormDataViewer
        formData={formData}
        onSelectFormNodeTypeData={formNodeTypeData =>
          navigation.push("FormNodeTypeDataViewerScreen", {
            ...route.params,
            formNodeTypeData: formNodeTypeData.summary,
          })
        }
      />
    </CenteredView>
  );
};
