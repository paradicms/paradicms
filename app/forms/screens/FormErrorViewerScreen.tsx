import * as React from "react";
import {FormStackScreenProps} from "../navigation/FormStackScreenProps";
import {View} from "react-native";
import {Text} from "@rneui/themed";

export const FormErrorViewerScreen: React.FunctionComponent<FormStackScreenProps<
  "FormErrorViewerScreen"
>> = ({route}) => {
  return (
    <View style={{flex: 1, alignItems: "center", justifyContent: "center"}}>
      <Text>{route.params.error.toString()}</Text>
    </View>
  );
};
