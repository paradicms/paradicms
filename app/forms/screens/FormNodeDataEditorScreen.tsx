import * as React from "react";
import {FormStackScreenProps} from "../navigation/FormStackScreenProps";
import {fontAwesomeIconFactory} from "../fontAwesomeIconFactory";
import {
  createFormPropertyDataEditor,
  FormNodeDataEditor,
} from "@paradicms/react-native-shacl-forms";
import {useFormData} from "../hooks/useFormData";

export const FormNodeDataEditorScreen: React.FunctionComponent<FormStackScreenProps<
  "FormNodeDataEditorScreen"
>> = ({navigation, route}) => {
  const {error, formData, saveFormData} = useFormData(route.params.formData.id);

  if (error) {
    navigation.navigate("FormErrorViewerScreen", {...route.params, error});
    return null;
  }

  if (!formData) {
    return null;
  }

  const formNodeTypeData = formData.nodeTypeById(
    route.params.formNodeTypeData.id
  );
  const formNodeData = formNodeTypeData.nodeById(route.params.formNodeData.id);

  return (
    <FormNodeDataEditor
      formPropertyDataEditorFactory={createFormPropertyDataEditor}
      formNodeData={formNodeData}
      iconFactory={fontAwesomeIconFactory}
      onChange={saveFormData}
    />
  );
};
