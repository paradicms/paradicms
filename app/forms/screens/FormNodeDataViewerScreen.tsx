import * as React from "react";
import {FormStackScreenProps} from "../navigation/FormStackScreenProps";
import {useFormData} from "../hooks/useFormData";
import {fontAwesomeIconFactory} from "../fontAwesomeIconFactory";
import {
  createFormPropertyDataViewer,
  FormNodeDataViewer,
} from "@paradicms/react-native-shacl-forms";

export const FormNodeDataViewerScreen: React.FunctionComponent<FormStackScreenProps<
  "FormNodeDataViewerScreen"
>> = ({navigation, route}) => {
  const {error, formData} = useFormData(route.params.formData.id);

  if (error) {
    navigation.navigate("FormErrorViewerScreen", {...route.params, error});
    return null;
  }

  if (!formData) {
    return null;
  }

  const formNodeTypeData = formData.nodeTypeById(
    route.params.formNodeTypeData.id
  );
  const formNodeData = formNodeTypeData.nodeById(route.params.formNodeData.id);

  return (
    <FormNodeDataViewer
      formNodeData={formNodeData}
      formPropertyDataViewerFactory={createFormPropertyDataViewer}
      iconFactory={fontAwesomeIconFactory}
    />
  );
};
