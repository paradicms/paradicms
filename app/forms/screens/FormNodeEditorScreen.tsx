import * as React from "react";
import {ScrollView} from "react-native";
import {FormStackScreenProps} from "../navigation/FormStackScreenProps";
import {useForm} from "../hooks/useForm";
import {FormNodeEditor} from "@paradicms/react-native-shacl-forms";
import {createFormPropertyEditor} from "@paradicms/react-native-shacl-forms/dist/editors/createFormPropertyEditor";
import {fontAwesomeIconFactory} from "../fontAwesomeIconFactory";

export const FormNodeEditorScreen: React.FunctionComponent<FormStackScreenProps<
  "FormNodeEditorScreen"
>> = ({navigation, route}) => {
  const {dispatchForm, form} = useForm();

  if (!form) {
    return null;
  }

  const formNodeType = form.nodeTypeById(route.params.formNodeTypeId);
  const formNode = formNodeType.nodeById(route.params.formNodeId);

  return (
    <ScrollView>
      <FormNodeEditor
        formPropertyEditorFactory={createFormPropertyEditor}
        formNode={formNode}
        iconFactory={fontAwesomeIconFactory}
        onChange={dispatchForm}
      />
    </ScrollView>
  );
};
