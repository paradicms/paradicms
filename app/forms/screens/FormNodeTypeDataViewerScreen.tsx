import * as React from "react";
import {FormStackScreenProps} from "../navigation/FormStackScreenProps";
import {useFormData} from "../hooks/useFormData";
import {FormNodeTypeDataViewer} from "@paradicms/react-native-shacl-forms";
import {CenteredView} from "../components/CenteredView";

export const FormNodeTypeDataViewerScreen: React.FunctionComponent<FormStackScreenProps<
  "FormNodeTypeDataViewerScreen"
>> = ({navigation, route}) => {
  const {error, formData} = useFormData(route.params.formData.id);

  if (error) {
    navigation.navigate("FormErrorViewerScreen", {
      ...route.params,
      formNodeData: null,
      error,
    });
    return null;
  }

  if (!formData) {
    return null;
  }

  const formNodeTypeData = formData.nodeTypeById(
    route.params.formNodeTypeData.id
  );

  return (
    <CenteredView>
      <FormNodeTypeDataViewer
        formNodeTypeData={formNodeTypeData}
        onSelectFormNodeData={formNodeData =>
          navigation.push("FormNodeDataViewerScreen", {
            ...route.params,
            formNodeData: formNodeData.summary,
          })
        }
      />
    </CenteredView>
  );
};
