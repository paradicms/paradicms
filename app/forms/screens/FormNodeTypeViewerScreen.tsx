import * as React from "react";
import {StyleSheet, View} from "react-native";
import {FormStackScreenProps} from "../navigation/FormStackScreenProps";
import {useForm} from "../hooks/useForm";
import {FormNodeTypeViewer} from "@paradicms/react-native-shacl-forms";

export const FormNodeTypeViewerScreen: React.FunctionComponent<FormStackScreenProps<
  "FormNodeTypeViewerScreen"
>> = ({navigation, route}) => {
  const {form} = useForm();

  if (!form) {
    return null;
  }

  const formNodeType = form.nodeTypeById(route.params.formNodeTypeId);

  return (
    <View style={styles.container}>
      <FormNodeTypeViewer
        formNodeType={formNodeType}
        onSelectFormNode={formNode =>
          navigation.navigate("FormNodeViewerScreen", {
            formNodeId: formNode.id,
            formNodeTypeId: formNodeType.id,
          })
        }
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
});
