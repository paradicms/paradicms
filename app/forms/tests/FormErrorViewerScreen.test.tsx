import * as React from "react";
import * as renderer from "react-test-renderer";
import {FormErrorViewerScreen} from "../screens/FormErrorViewerScreen";

test("renders correctly", () => {
  const tree = renderer
    .create(
      <FormErrorViewerScreen
        // @ts-ignore
        navigation={{push: jest.fn()}}
        // @ts-ignore
        route={{params: {error: "test"}}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
