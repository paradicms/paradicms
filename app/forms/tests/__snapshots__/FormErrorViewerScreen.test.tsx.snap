// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessibilityRole="text"
    style={
      {
        "color": "#242424",
      }
    }
  >
    test
  </Text>
</View>
`;
